src/Lean/Elab/App.lean

```diff
@@ -769,15 +769,24 @@ private partial def elabAppFnId (fIdent : Syntax) (fExplicitUnivs : List Level)
     (namedArgs : Array NamedArg) (args : Array Arg) (expectedType? : Option Expr) (explicit ellipsis overloaded : Bool) (acc : Array TermElabResult)
     : TermElabM (Array TermElabResult) := do
   match fIdent with
-  | Syntax.ident _ _ n preresolved =>
+  | Syntax.ident _ sstr n preresolved =>
     let funLVals ← withRef fIdent $ resolveName n preresolved fExplicitUnivs
     let overloaded := overloaded || funLVals.length > 1
     -- Set `errToSorry` to `false` if `funLVals` > 1. See comment above about the interaction between `errToSorry` and `observing`.
     withReader (fun ctx => { ctx with errToSorry := funLVals.length == 1 && ctx.errToSorry }) do
       funLVals.foldlM (init := acc) fun acc ⟨f, fields⟩ => do
-        unless lvals.isEmpty && args.isEmpty && namedArgs.isEmpty do
+        let (fSstr, fieldSstrs) := fields.foldr (init := (sstr, [])) fun fieldName (restSstr, ss) =>
```

Comment:
@leodemoura I didn't see an immediate way to get the source info for `.foo` projections since `resolveName` only returns `List String`, so I added this traversal. Happy to revert if there is a better way.

Reply:
I will try to add modify `resolveName`. It feels odd to lose the information there, and have to reconstruct it here.

Reply:
I added `resolveName'` and used it here to get precise position information. `resolveName'` is basically implementing the logic you wrote here.

Reply:
https://github.com/leanprover/lean4/commit/ae60360d7abfa31c4a9ccdc9eb8e362d9a82ed7c

Reply:
Thank you! There was an issue with the logic on French-quoted fields as seen [here](https://github.com/leanprover/lean4/pull/260/checks?check_run_id=1710379702#step:14:441) where the field ("end") is not the same as the written-field ("«end»"). An approach which seems to work is:
```lean
let (fSstr, fieldSstrs) := fields.foldr (init := (sstr, [])) fun _ (restSstr, ss) =>
  let fieldSstr := restSstr.takeRightWhile fun c => c ≠ '.'
  ({ restSstr with startPos := restSstr.startPos - (fieldSstr.bsize + 1) }, fieldSstr :: ss)
```